src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java

```diff
@@ -142,6 +146,109 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload d
     return future;
   }
 
+  private Future<Object> setApprovedFalseIfUserNotHaveApprovalPermission(DataImportEventPayload dataImportEventPayload, JsonObject tenantConfig, RequestContext requestContext) {
+    CompositePurchaseOrder order = Json.decodeValue(dataImportEventPayload.getContext().get(ORDER.value()), CompositePurchaseOrder.class);
+    Boolean isApproved = order.getApproved();
+    boolean isApprovalRequired = PurchaseOrderHelper.isApprovalRequiredConfiguration(tenantConfig);
+    boolean isUserNotHaveApprovalPermission = PurchaseOrderHelper.isUserNotHaveApprovePermission(requestContext);
+
+    if (isApprovalRequired && Boolean.TRUE.equals(isApproved) && isUserNotHaveApprovalPermission) {
+      order.setApproved(false);
+      dataImportEventPayload.getContext().put(ORDER.value(), Json.encode(order));
+    }
+    return Future.succeededFuture();
+  }
+
+  private Future<Void> adjustEventType(DataImportEventPayload dataImportEventPayload, JsonObject tenantConfig, Map<String, String> okapiHeaders) {
+    CompositePurchaseOrder order = Json.decodeValue(dataImportEventPayload.getContext().get(ORDER.value()), CompositePurchaseOrder.class);
+    WorkflowStatus workflowStatus = extractWorkflowStatus(dataImportEventPayload);
+
+    Promise<Void> promise = Promise.promise();
+
+    if (workflowStatus.equals(WorkflowStatus.PENDING)) {
+      LOGGER.debug("adjustEventType:: set event type DI_ORDER_CREATED for jobExecutionId {}", dataImportEventPayload.getJobExecutionId());
+      return Future.succeededFuture();
+    }
+
+    Boolean isApproved = order.getApproved();
+    boolean isApprovalRequired = PurchaseOrderHelper.isApprovalRequiredConfiguration(tenantConfig);
+
+    if (workflowStatus.equals(WorkflowStatus.OPEN)) {
+      if (Boolean.FALSE.equals(isApproved) && isApprovalRequired) {
+        LOGGER.debug("adjustEventType:: set event type DI_ORDER_CREATED for jobExecutionId {}", dataImportEventPayload.getJobExecutionId());
```

Comment:
Hello, if approval is required and the user does have not permission, then we need to send DI_COMPLETED, otherwise, if approval is not required, we need to send DI_PENDING_ORDER_CREATED

Reply:
Hello, @RuslanLavrov, if approval is required and the user does have not permission, then we need to send DI_COMPLETED, otherwise, if approval is not required, we need to send DI_PENDING_ORDER_CREATED